node/cli/cli.go

```diff
@@ -0,0 +1,108 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/internal/commands/providers"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/internal/commands/root"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/internal/commands/version"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/provider"
+)
+
+// CommandOption sets an option on the command.
+type CommandOption func(*Command)
+
+// Command builds the CLI command
+type Command struct {
+	cmd        *cobra.Command
+	s          provider.Store
+	name       string
+	setupTrace TracingSetupFunc
+}
+
+// WithProvider registers a provider which the cli can be initialized with.
+func WithProvider(name string, f provider.InitFunc) CommandOption {
+	return func(c *Command) {
+		if c.s == nil {
+			c.s = provider.NewStore()
```

Comment:
Why do you use the store still? It seems weird to continue to support multiple providers?

Reply:
Yeah, I debated removing it...
Really it'd be nice to be able to use this at least to switch beetwen mockv0 and mockv1 (async).
But I guess we could just as easily create two CLI's.

Reply:
Or just one CLI with a flag `--mode={sync,async}` that defaults to whatever is preferred - I'd argue sync is as it's been around for longer.

Reply:
Perhaps mode is more suitable as a provider specific config passed through the config file.